sdk/src/mcm_dp.c

```diff
@@ -6,17 +6,120 @@
 #include <bsd/string.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <errno.h>
 
 #include "memif_impl.h"
 #include "udp_impl.h"
 #include "logger.h"
 #include "mcm_dp.h"
 #include "media_proxy_ctrl.h"
 
+/* Calculate audio buffer size based on ST2110-30 related parameters */
+static int mcm_calc_audio_buffer_size(mcm_audio_args *params, uint32_t *size)
+{
+    uint32_t sample_size = 0;
+    uint32_t sample_num = 0;
+
+    switch (params->format) {
+    case AUDIO_FMT_PCM8:
+        sample_size = 1;
```

Comment:
Is there any way to automate this formula somehow? Dependency between format and sample_size value?

Reply:
https://github.com/OpenVisualCloud/Media-Communications-Mesh/blob/dde0aaea3cb2d26ac5cac135133bda303814c818/sdk/include/mcm_dp.h#L112-L118
Perhaps:
```c
sample_size = params->format + 1;
```

Reply:
> Is there any way to automate this formula somehow? Dependency between format and sample_size value?

This function is implemented the same way as in MTL. Take a look https://github.com/OpenVisualCloud/Media-Transport-Library/blob/3127c88b8a73d59520189c72254559a436125ed3/lib/src/st2110/st_fmt.c#L1037

I suggest to leave it as it is in order to stay aligned with MTL implementation.

Reply:
> https://github.com/OpenVisualCloud/Media-Communications-Mesh/blob/dde0aaea3cb2d26ac5cac135133bda303814c818/sdk/include/mcm_dp.h#L112-L118
> 
> 
> Perhaps:
> ```c
> sample_size = params->format + 1;
> ```

I can't agree with that. You're suggesting to remove clarity from the code. A golden rule here is "Do not optimize prematurely".